---
layout: post
title: "Cramer's V vs F<sub>ST</sub> for Insect Population Genetics"
date: 2017-04-04 00:01:19
categories: "bioinformatics"
tags: ["statistics"]
---
[Fixation index](https://en.wikipedia.org/wiki/Fixation_index), or F<sub>ST</sub>, is a univariate statistic calculated as the ratio of variance within populations to the variance between populations. Within insect population genetics, F<sub>ST</sub> is used to score, and then rank, the correlation between variants and the population structure.

The focus of my Ph.D. dissertation was to investigate variable importance measures as calculated via Random Forests as an alterative to F<sub>ST</sub>.  I've also begun looking at [Logistic Regression Ensembles](http://rnowling.github.io/machine/learning/2017/02/14/lr-gwas.html).

In addition to these two machine learning approaches, I wanted to investigate a statistical method, Cramer's V.  [Cramer's V](https://en.wikipedia.org/wiki/Cram%C3%A9r's_V) measures the assocation (correlation of unsigned variables) of nominal (categorical) variables.  I went ahead and implemented Cramer's V in the `dev` branch of my population methods exporation toolkit [Asaph](https://github.com/rnowling/asaph/).

I used the Burkina Faso *An. gambiae* and *An. coluzzii* samples from the [Anopheles gambiae 1000 genomes project](https://www.malariagen.net/projects/ag1000g) to compare Cramer's V and F<sub>ST</sub>.  I calculated the F<sub>ST</sub> scores for each SNP using [vcftools](https://vcftools.github.io/).  I calculated Cramer's V using Asaph on data imported using both the counts and categories feature encoding schemes.  I then plotted F<sub>ST</sub> vs Cramer's V (counts) and F<sub>ST</sub> vs Cramer's V (categories) to get a sense of the correlation between the two metrics.

![Fst vs Cramer's V (counts)](/images/cramers-v-vs-fst/bfm_vs_bfs_fst_vs_cramers_v_counts.png)

![Fst vs Cramer's V (categories)](/images/cramers-v-vs-fst/bfm_vs_bfs_fst_vs_cramers_v_categories.png)

The above figures give the scatter plots of F<sub>ST</sub> vs Cramer's V with the counts and categories feature encodings, respectively. Cramer's V calculated on the count-encoded features has a $$r^2$$ value of 0.865 vs F<sub>ST</sub>, while Cramer's V calculated on the count-encoded features has a $$r^2$$ value of 0.818 vs F<sub>ST</sub>.


## Conclusion
Along with Random Forests and Logistic Regression Ensembles, Cramer's V is another alternative to F<sub>ST</sub> for finding variants that best describe the genetic basis of differences between two populations.  Cramer's V correlates well with F<sub>ST</sub>, but a simple correlation analysis doesn't tell us which metric is more appropriate for a given situation.  Substantial work remains to validate the four methods and compare them.

My current hypothesis, though, is that no particular method (F<sub>ST</sub>, Random Forests, Logistic Regression, and Cramer's V) will give substantially better results than the others.  They are all trying to fundamentally solve the same problem.

When looking through the top-rated SNPs produced by each method, it's striking that a number of those SNPs are non-functional.  They lie in introns or far removed from a gene or its promoter region.  These methods only take into account the variants' abilities to distinguish the populations, but not whether the variants are likely to lead to phenotypic differences.

Going forward, I think there is value in focusing on methods that incorporate additional information about the biological systems.  For example, a recent paper describes a method [cepip](http://genomebiology.biomedcentral.com/articles/10.1186/s13059-017-1177-3) that attempts to determine if variants in humans are pathogenic or not.  We could do something similar for insect population genetics by incorporating information about the neighbors of the variants such as the GC content, average dN/dS if they are located within a gene, or how well conserved neighboring nucleotides are.  At the very least, we may get an automated way to filter out obviously non-functional variants.  More optimistically, we may get an approach that can reliably estimate how likely a variant is to be functional and rank more biologically-interesting variants higher.
